src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -848,6 +845,24 @@ abstract class BTypes {
    * infos using `get`, but it reports inliner warnings for missing infos that prevent inlining.
    */
   final case class ClassBType(internalName: InternalName) extends RefBType {
+
+    override def descriptor: String = {
+      if (descriptorCache eq null) {
+        val sb = new java.lang.StringBuilder(internalName.length + 2)
+        buildDescriptor(sb)
```

Comment:
Yes it would be concurrent, but I believe that it is threadsafe

I don't believe that the cost of lock is worth the potential gain in lost CPU

the answer is the same (==) so 2 threads race one will win. Small chance that we will have 2 thread with identical strings, but this is not harmful

There is no chance of a incomplete string being visible to another thread as String constructor has a memory fence